build_tools/packaging/promote_from_rc_to_final.py

```diff
@@ -0,0 +1,278 @@
+import argparse
+from packaging.version import Version
+import pathlib
+from pkginfo import Wheel
+import glob
+import sys
+import tempfile
+import tarfile
+import fileinput
+import os
+import glob
+import importlib.util
+
+# Need dynamic load as change_wheel_version needs to be imported via parent directory
+this_file = pathlib.Path(__file__).resolve()
+build_tools_dir = this_file.parent.parent
+# ../third_party/change_wheel_version/change_wheel_version.py
+change_wheel_version_path = (
+    build_tools_dir / "third_party" / "change_wheel_version" / "change_wheel_version.py"
+)
+
+spec = importlib.util.spec_from_file_location(
+    "third_party_change_wheel_version", change_wheel_version_path
+)
+change_wheel_version = importlib.util.module_from_spec(spec)
+assert spec.loader is not None
+spec.loader.exec_module(change_wheel_version)
+
+assert hasattr(
+    change_wheel_version, "change_wheel_version"
+), "change_wheel_version module does not expose change_wheel_version function"
+
+
+def parse_arguments(argv):
+    parser = argparse.ArgumentParser(
+        description="""Promotes packages from release candidate to final release (e.g. 7.10.0rc1 --> 7.10.0).
+
+Promotion works for for wheels and .tar.gz.
+Wheels version is determined by python library to interact with the wheel.
+For tar.gz., the version is extract from <.tar.gz>/PKG-INFO file.
+"""
+    )
+    parser.add_argument(
+        "--input_dir",
+        help="Path to the directory that contains .whl and .tar.gz files to promote",
+        type=pathlib.Path,
+    )
+    parser.add_argument(
+        "--match-files",
+        help="Limits selection in '--input-dir' to files matchings this argument. Use wild cards if needed, e.g. '*rc2*' (default '*' to promote all files in '--input-dir')",
+        default="*",
+    )
+    parser.add_argument(
+        "--delete",
+        help="Deletes old file after successful promotion",
+        action="store_true",
+    )
+    return parser.parse_args(argv)
+
+
+def wheel_change_extra_files(new_dir_path, old_version, new_version):
+    # extract "rocm_sdk_core" from /tmp/tmp3swrl25j/wheel/rocm_sdk_core-7.10.0
+    package_name_no_version = (
+        str(new_dir_path).rsplit("/", 1)[-1].split(str(new_version))[0][:-1]
+    )
```

Comment:
If `new_dir_path` is a `Path`, maybe this can use `.name` (https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.name) instead of `str(new_dir_path).rsplit("/", 1)[-1]`?